src/oidc/oidc.ts

```diff
@@ -311,3 +311,80 @@ export const OAuth2Logout = (WSLogoutAndRedirect: () => void) => {
         cleanup();
     };
 };
+
+/**
+ * Logs out the user from the OIDC provider using the SignOutRedirect function of the UserManager.
+ * @param {RequestOidcAuthenticationOptions} options - Configuration options for the OIDC logout request
+ * @param {string} options.redirectCallbackUri - The callback page URI to redirect back
+ * @param {string} options.postLogoutRedirectUri - The URI to redirect after successfully logging out
+ * @throws {Error} If the ID token is missing in session storage
+ */
+export const oidcLogout = async (options: RequestOidcAuthenticationOptions): Promise<void> => {
+    try {
+        const { redirectCallbackUri, postLogoutRedirectUri } = options;
+
+        const { appId } = getServerInfo();
+
+        const oidcConfig = JSON.parse(localStorage.getItem('config.oidc_endpoints') || '{}');
+        const serverUrl = oidcConfig.issuer;
+
+        const sessionStorageKey = `oidc.user:${serverUrl}:${appId}`;
+        const getSessionStorageOIDCUser = JSON.parse(window.sessionStorage.getItem(sessionStorageKey) || '{}');
+
+        if (!getSessionStorageOIDCUser.id_token) {
+            throw new Error('ID token is missing in session storage.');
+        }
+
+        const userManager = await createUserManager({
+            redirectCallbackUri,
+            postLogoutRedirectUri,
+        });
+
+        await userManager.signoutRedirect({
```

Comment:
Anything after this function will not get executed since we are redirecting to the `post_logout_redirect_uri`